pkg/orchestration/wiring/postgres/plugin.go

```diff
@@ -75,43 +74,13 @@ func (p *WiringPlugin) WireUp(resource *orch_v1.StateResource, context *wiringpl
 		smith.DeletionDelayAnnotation: deletionDelay.String(),
 	}
 
-	instanceParameters, err := instanceParameters(resource, context)
-	if err != nil {
-		return nil, false, err
-	}
-	serviceInstance.Spec.Parameters = &runtime.RawExtension{
-		Raw: instanceParameters,
-	}
-
-	references, err := references(context)
-	if err != nil {
-		return nil, false, err
-	}
-
-	serviceInstanceResource := smith_v1.Resource{
-		Name:       wiringutil.ServiceInstanceResourceName(resource.Name),
-		References: references,
-		Spec: smith_v1.ResourceSpec{
-			Object: serviceInstance,
-		},
-	}
-
-	shapes, err := shapes(resource, &serviceInstanceResource, context)
+	// if this postgres depends on Dedicated RDS, it means it should be created there instead on the Default RDS
+	// there should be only one RDS dependency
+	sharedDbDep, found, err := context.FindTheOnlyDependency()
```

Comment:
This functions now has a single place where we are looking for the only dependency.